vlmeval/vlm/emu.py

```diff
@@ -0,0 +1,87 @@
+import torch
+from PIL import Image
+from abc import abstractproperty
+import os.path as osp
+import os 
+from ..smp import *
+
+class Emu:
+
+    def __init__(self, 
+                 name, 
+                 model_path_map={
+                     "emu2":"BAAI/Emu2",
+                     "emu2_chat":"BAAI/Emu2_Chat"
+                     }, 
+                 **kwargs):
+        
+        self.model_path_map = model_path_map
+        assert name in self.model_path_map or osp.exists(name) or splitlen(name) == 2
+        if name in self.model_path_map:
+            model_path = self.model_path_map[name]
+        else:
+            model_path = name
+
+        assert osp.exists(model_path) or splitlen(model_path) == 2
+        
+        from transformers import AutoModelForCausalLM, AutoTokenizer
+        tokenizer = AutoTokenizer.from_pretrained(model_path) # "BAAI/Emu2-Chat"
+        self.tokenizer = tokenizer
+        self.device = torch.device("cuda") if torch.cuda.is_available() else "cpu"
+        device = self.device
+        model = AutoModelForCausalLM.from_pretrained(
+            model_path, # "BAAI/Emu2-Chat"
+            torch_dtype=torch.bfloat16,
+            low_cpu_mem_usage=True,
+            trust_remote_code=True).to(device).eval()
+        self.model = model
+        self.kwargs = {'max_length': 128}
+        
+        
+    def generate(self, image_path, prompt, dataset=None):
```

Comment:
You can use interleave_generate to implement generate, seems they can be formulates to the same form.